packages/esm-patient-registration-app/src/patient-registration/field/obs/obs-field.component.tsx

```diff
@@ -159,15 +159,38 @@ interface DateObsFieldProps {
   concept: ConceptResponse;
   label: string;
   required?: boolean;
-  dateFormat?: string;
-  placeholder?: string;
+  minDate?: string;
+  maxDate?: string;
 }
 
-function DateObsField({ concept, label, required, placeholder }: DateObsFieldProps) {
+const evaluateConfigDate = (dateString: string): Date | string | null => {
+  if (!dateString) {
+    return null;
+  }
+
+  if (dateString === 'today') {
+    return new Date();
+  }
+
+  const date = new Date(dateString);
+  if (!isNaN(date.getTime())) {
+    return date;
+  }
+
+  return 'Invalid min or max date!';
```

Comment:
1. This is a user-facing string so it must be localizable
2. Make the message friendlier by passing in as an argument whether this is the min date or max date
3. Instead of checking the return type, throw an error and use try...catch. It may be better to actually create the error message at that point.